.github/workflows/tests.yml

```diff
@@ -35,12 +35,54 @@ jobs:
       - name: Coverage reporting
         run: |
           coverage combine
-          coverage report -m
-          coverage xml
-          coverage html
-      - name: Publish coverage results
-        uses: codecov/codecov-action@v4
+          coverage report --show-missing
+          mv .coverage .coverage.${{ matrix.python-version }}
+      - name: Store coverage file
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-${{ matrix.python-version }}
+          path: .coverage.${{ matrix.python-version }}
+
+  coverage:
+    name: Coverage
+    runs-on: ubuntu-latest
+    needs: tests
+    permissions:
+      # If the author is a maintainer, the permission level is set by the
+      # values below.
+      # `pull-requests: write` is needed for publishing new comments in pull
+      # requests.
+      # `contents: write` is needed for pushing data to the
+      # `python-coverage-comment-action` branch, and for editing existing
+      # comments (to avoid publishing multiple comments in the same PR)
+      # In case the pull request comes from a forked repository, the maximum
+      # permission level is read, so the permissions below won't be acted upon
+      # by GitHub.
+      # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token
+      pull-requests: write
+      contents: write
+    steps:
+      - uses: actions/checkout@v4
 
+      - uses: actions/download-artifact@v4
+        id: download
+        with:
+          pattern: coverage-*
+          merge-multiple: true
+
+      - name: Coverage comment
+        id: coverage_comment
+        uses: py-cov-action/python-coverage-comment-action@v3
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          MERGE_COVERAGE_FILES: true
```

Comment:
I haven't changed the action settings more than necessary. In particular, if you want your badge to appear green, you'll have to tweak `MINIMUM_GREEN` which by default is `100` (which might be a bit overkill). Your current coverage is 98% (👏) and will appear orange with the default settings.